sagemathinc/cocalc
# webapp: idea for loading feedback

# Description

I needed a distraction.

This is a proof of concept for a webapp client loading indicator. What's missing is proper error handling, testing, and bugfixing. Still, it's nice to see some activity. Here is how it looks with a simulated slow network connection. If you want to fine tune it, you're welcome. E.g. with that, the animations are no longer that interesting …

![Peek 2020-07-03 12-25](https://user-images.githubusercontent.com/207405/86461752-be777f80-bd2a-11ea-88a0-f719ad7db2e6.gif)

# Testing Steps
well, obviously across some browsers. there is no actual code change of the client code, so, if it passes the banner page it is all good. browserstack could be helpful …

what I checked:
* throwing an arbitrary error inside the load progress or somewhere else is caught. interestingly, this really needs an additional `.catch(...)`
* firefox and chrome: no problems on my windows machine
* browserstack:
  - win 7: edge 83 OK, IE 11 blocked as it shouldbe, Safari 5 blocked as well, Chrome 83 OK, older Firefox 67 OK
  - win 10: edge 80 OK, chrome 79 OK
  - android pixel 3xl: chrome OK (and well, I can see that on mobile these progress bars are funny, but I'm ignoring this for now)
  - samsung galaxy 10 (android 9) chrome: ok
  - Mac Catalina: Safari 13.1 OK, chrome 83 OK
  - mac Mojave: Safari 12.1 OK
  - mac high sierra: Safari 11.1 OK
  - iOS IPhone 11: Safari ?? OK
  - iOS IPad Pro 11 2018: Safari ?? OK
  - IOS IPad 12.9 2018: Chrome ?? OK

So, long story short, I tried a bunch of combinations and the landing page with the sign up/in forms did show up all the time.

# Relevant Issues

### [Checklist](https://github.com/sagemathinc/cocalc/wiki/PR-Checklist):

Front end:
